### PR TITLE
[safety-rules] test the binary

### PR DESCRIPTION
**Pull request by davidiw**
This removes the nonsense that was SpawnedProcess and replaces it with a test to ensure that our binary loads -- which was the only purpose of SpawnedProcess.

As a follow up, we should consider making a test-utils crate for consensus with the hope that we can eliminate some of the testing flags in our code base.

### CODE REVIEW

**Comment by bmwill** [comment]
I can't leave a comment on it, but it looks like this adds an empty file `consensus/safety-rules/tests/stupid_test.rs`?

File: consensus/safety-rules/tests/binary.rs [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+const BINARY: &str = env!("CARGO_BIN_EXE_safety-rules");
+
+#[test]
+fn test_consensus_state() {
+    let mut config = NodeConfig::random().consensus.safety_rules;
+    let test_config = config.test.as_mut().unwrap();
+    let private_key = test_config.consensus_key.as_ref().unwrap().private_key();
+    let signer = ValidatorSigner::new(test_config.author, private_key);
+    let waypoint = test_utils::validator_signers_to_waypoint(&[&signer]);
```
Comment (bmwill) [inline consensus/safety-rules/tests/binary.rs:20]: Is there no way to get this information without this particular test_util? If so then you could drop the requirement to build with the testing feature flag

**Comment by davidiw** [comment]
> I can't leave a comment on it, but it looks like this adds an empty file `consensus/safety-rules/tests/stupid_test.rs`?

This was part of my 45 minute exploration into why my integration test wouldn't compile :D

File: consensus/safety-rules/tests/binary.rs (hunk 1 above)
Comment (davidiw) [reply consensus/safety-rules/tests/binary.rs:20]: I mean we could generate these somehow else by copypastaing stuff around, but these values are required to start the binary and keep it running.

**Comment by davidiw** [comment]
> you know if you cleanup LSR you have to cleanup LEC too right?

begrudgingly

File: consensus/safety-rules/tests/binary.rs (hunk 1 above)
Comment (bmwill) [reply consensus/safety-rules/tests/binary.rs:20]: I'm specifically talking about this single function here, `test_utils::validator_signers_to_waypoint`. But yeah i understand.

**Comment by bmwill** [comment]
I mean I had a diff out to fix both, but David decided he wanted to do it in a better way :D

**Pull request closed without merging**